File: spring-grpc-server/server/src/main/java/com/example/springgrpc/server/GrpcClientConfiguration.java
package com.example.springgrpc.server;

import com.example.springgrpc.client.HelloGrpcClient;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class GrpcClientConfiguration {
    @Bean(destroyMethod = "close")
    public HelloGrpcClient helloGrpcClient(GrpcClientProperties properties) {
        return HelloGrpcClient.connect(
            properties.getHost(),
            properties.getPort(),
            properties.getTimeoutSeconds()
        );
    }
}

File: spring-grpc-server-kotlin/server/src/main/kotlin/com/example/springgrpc/server/GrpcClientConfiguration.kt
package com.example.springgrpc.server

import com.example.springgrpc.client.HelloGrpcClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class GrpcClientConfiguration {
    @Bean(destroyMethod = "close")
    fun helloGrpcClient(properties: GrpcClientProperties): HelloGrpcClient =
        HelloGrpcClient.connect(
            host = properties.host,
            port = properties.port,
            timeoutSeconds = properties.timeoutSeconds,
        )
}
